swagger.yaml

```diff
@@ -0,0 +1,350 @@
+      description: |
+        Update the state of the import job. The following states are valid.
+        * Created - The job has been created (Default state when created)
+        * Inprogress - The dataset is being converted and ingested into a store
+        * Completed - The dataset has been completed
+        * Failed - The dataset has failed to be completed. See the events messages
+      parameters:
+      - $ref: '#/parameters/instanceId'
+      - $ref: '#/parameters/state'
+      produces:
+      - "application/json"
+      security:
+      - InternalAPIKey: []
+      responses:
+        200:
+          description: "The import state has been updated"
+        404:
+          description: "InstanceId does not match any import jobs"
+        500:
+          description: "Failed to process the request due to an internal error"
+  /import/{instanceId}/dimensions:
+    put:
+      tags:
+      - "Internal Interface"
+      summary: "Add a dimension for the import job"
+      description: |
+        Add a dimension from the obervations found in the V3 data. To create a dimension the following items are needed;
+        * nodeName - The name of the node eg (instanceId)-(dimension).(value)
+        * nodeId - The unique id for this node
+      parameters:
+      - $ref: '#/parameters/instanceId'
+      - $ref: '#/parameters/dimension'
+      produces:
+      - "application/json"
+      security:
+      - InternalAPIKey: []
+      responses:
+        200:
+          description: "The dimension was added"
+        404:
+          description: "InstanceId does not match any import jobs"
+        500:
+          description: "Failed to process the request due to an internal error"
+    get:
+      tags:
+      - "Internal Interface"
+      summary: "Get all dimensions from an imported job"
+      description: "Get all the dimensions within an import job"
+      parameters:
+      - $ref: '#/parameters/instanceId'
+      produces:
+      - "application/json"
+      security:
+      - InternalAPIKey: []
+      responses:
+        200:
+          description: "Return a list of dimensions"
+          schema:
+            $ref: '#/definitions/Dimensions'
+        404:
+          description: "InstanceId does not match any import jobs"
+        500:
+          description: "Failed to process the request due to an internal error"
+definitions:
+  JobInstance:
+    type: object
+    description: "A object returned when an import job is created"
+    properties:
+      instanceId:
+        type: string
+        description: "A unique id for an import job"
+  ImportJob:
+    type: object
+    description: "A object used to create a import job"
+    properties:
+      dataset:
+        type: string
+        description: "The name of the dataset"
+      edition:
+        type: string
+        description: "The edition of the dataset"
+      version:
+        type: string
+        description: "The version of the dataset"
+      S3Url:
+        type: array
```

Comment:
This may need to become a map instead of an array. We need to be able to identify which input file is which.